src/dso_api/dynamic_api/serializers.py

```diff
@@ -636,6 +676,92 @@ def _build_serializer_embeddded_field(
     new_attrs[camel_name] = embedded_field
 
 
+def _through_serializer_factory(
+    model: Type[DynamicModel], target_model: DynamicModel, target_field_name: str
+) -> Type[DynamicSerializer]:
+    """Generate the DRF serializer class for a M2M model.
+
+    Args:
+        model: the `through` model
+        target_model: the target model for the M2M relation
+        target_field_name: the name of the M2M field in the source model
+
+        This factory adds a field `target_field_name`. A field with this
+        name is available on the through model. It is a FK pointing to the target.
+
+        This FK field will lead to a HALTemporalHyperlinkedRelatedField on the
+        through serializer. Because we do not want these field as subfields,
+        the ThroughSerializer is merging the ouput of this field
+        in its `to_representation` method.
+
+        Furthermore, temporal fields are add to the through serializer
+        if the target model is temporal.
+
+        In the creation of the ThroughSerializer, a special attribute `target`
+        is added to the `Meta` object. This is needed by the ThroughSerializer
+        to be able to pick the right field for embedding into its output.
+    """
+    fields = []
+    extra_kwargs = {}
+
+    safe_dataset_id = to_snake_case(model.get_dataset_id())
+    serializer_name = f"{safe_dataset_id.title()}{model.__name__}ThroughSerializer"
+    new_attrs = {
+        "table_schema": model.table_schema(),
+        "__module__": f"dso_api.dynamic_api.serializers.{safe_dataset_id}",
+    }
+
+    # The FK pointing to the target
+    fields.append(target_field_name)
+
+    if target_model.is_temporal():
+
+        table_schema = target_model.table_schema()
+        temporal: Temporal = cast(Temporal, table_schema.temporal)
+
+        # The fields that define the boundaries of a particular related object are
+        # added if they exist on the model.
+        # (e.g. "beginGeldigheid" and "eindGeldigheid" for the "geldigOp" dimension
+        # for GOB data)
+        # NB. the `Temporal` dataclass return the boundary_fieldnames as snakecased!
+        for dimension_fieldname, boundary_fieldnames in temporal.dimensions.items():
+            for fn in boundary_fieldnames:
+                snaked_fn = to_snake_case(fn)
+                camel_fn = toCamelCase(fn)
+                if hasattr(model, snaked_fn):
```

Comment:
Misschien beter expliciet checken of een naam ook als veld voorkomt in de `model._meta.get_fields()` ?

Reply:
Eens, dat is cleaner dan een `hasattr`, pas ik aan.